pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -359,6 +362,34 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 			},
 			Sni: tls.Sni,
 		}
+	case networking.TLSSettings_ISTIO_MUTUAL:
+		// TODO(incfly): needs to change when SDS is ready.
+		cluster.TlsContext = &auth.UpstreamTlsContext{
```

Comment:
Maybe change the authN plugin function to a regular function to set these config and call it here. With that, the code that related to Istio-mTLS still stays closed to each other.